### PR TITLE
Symfony Flex support

### PR DESCRIPTION
**Pull request by mdriessen**
I just finished support for Symfony Flex projects! Symfony Flex is optional in Symfony 3.4 and used by default in Symfony 4.

As these changes break compatibility with Symfony projects not using Flex, a major version bump is required. I would suggest just retagging the 3.0.0 commit or using 3.1.

If you can create a branch for the current 2.0.0 version i'll submit another pull request that fixes the deprecations notices for Symfony 3.4 projects not using Flex. Then people should use ^2.0 for non-Flex projects and ^3.0 for Flex projects.

One task remains and that is submitting a PR to https://github.com/symfony/recipes-contrib so that this bundle can be automatically configured. If you like these changes and merge them, i'll open a pull request over there with your permission.

### CODE REVIEW

**Comment by mdriessen** [comment]
@shivas I think I'm done with changing things now ;) 

Apart from making the bundle work with Symfony 4, I also adopted some new design principles in making things simpler. From the installation to configuring custom providers and formatters, it feels more like a bundle made for Symfony 4.

Could you review the changes and tell me what you think about them? Thanks!

**Comment by shivas** [comment]
Thanks for PR, i'll review soon(tm) 👍

File: README.md [hunk 1]
```diff
@@ -28,96 +29,51 @@ Providers implemented:
...
+Console commands
+-
+
+There are two available console commands. The app:version:bump command is automatically called by Composer on every install and update.
 ```
+# This will display all available version providers
+bin/console app:version:list-providers
 
-The downside is that the version will be computed every time a twig layout is loaded, even if the variable is not used in the template. However, it could be useful if you have rapid succession of new versions or if you are afraid to forget to bump the version.
+# Display a dry run of a version bup
```
Comment (shivas) [inline README.md]: "bup" should be "bump"

**Comment by mdriessen** [comment]
That was sooner(tm) than I expected haha. Already caught and fixed the typo in my last commit.

There is indeed a big BC break and since 3.0 doesn't work with Symfony 4 as it stands it sounds more logical to me to just release a 3.1 version instead of a 4.0. In the Symfony Flex recipe I can specify a minimum requirement of 3.1 so it doesn't effect 3.0.

Could you also make a branch for the current 2.0.0 version (or is that the alpha branch?). I'll fix the 2 deprecation notices in that branch for Symfony 3.4 users that have not yet migrated to Flex.

**Comment by shivas** [comment]
I have pushed 2.0.0 branch based on 2.0.0 tag, so you are good to add PR to that branch

**Pull request merged**